src/filters/filter_chain.cc

```diff
@@ -16,16 +19,32 @@
 namespace authservice {
 namespace filters {
 
-FilterChainImpl::FilterChainImpl(config::FilterChain config,
+FilterChainImpl::FilterChainImpl(boost::asio::io_context& ioc,
+                                 config::FilterChain config,
                                  unsigned int threads)
     : threads_(threads),
       config_(std::move(config)),
-      oidc_session_store_(nullptr) {}
+      oidc_session_store_(nullptr) {
+  for (auto& filter : config_.filters()) {
+    if (filter.oidc().jwks_config_case() == config::oidc::OIDCConfig::kJwks) {
+      jwks_storage_map_.emplace_back(
+          std::make_shared<oidc::PermanentJwksStorageImpl>(
+              filter.oidc().jwks(), google::jwt_verify::Jwks::Type::JWKS));
+    } else {
+      jwks_storage_map_.emplace_back(
+          std::make_shared<oidc::NonPermanentJwksStorageImpl>(
+              filter.oidc().jwks_fetcher().jwks_uri(),
+              std::chrono::seconds(
+                  filter.oidc().jwks_fetcher().request_duration_sec()),
```

Comment:
is the default value is 0 if not specified? shall we have some sanity check to make a minimum, say 600 seconds?

for reference, istiod default jwks update is 20min. https://istio.io/latest/docs/reference/commands/pilot-discovery/. search for "PILOT_JWT_PUB_KEY_REFRESH_INTERVAL".